android-core/src/main/kotlin/com/mparticle/internal/ApplicationContextWrapper.kt

```diff
@@ -0,0 +1,351 @@
+package com.mparticle.internal
+
+import android.annotation.SuppressLint
+import android.annotation.TargetApi
+import android.app.Activity
+import android.app.Application
+import android.content.ComponentCallbacks
+import android.content.Context
+import android.content.res.Configuration
+import android.os.Build
+import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import com.mparticle.MParticle
+import java.lang.ref.WeakReference
+import java.util.Collections
+import java.util.LinkedList
+
+open class ApplicationContextWrapper(private val mBaseApplication: Application) : Application() {
+    var isReplayActivityLifecycle: Boolean = true
+    private var mRecord = true
+    private var mActivityLifecycleCallbackRecorder: ActivityLifecycleCallbackRecorder?
+
+    enum class MethodType {
+        ON_CREATED, ON_STARTED, ON_RESUMED, ON_PAUSED, ON_STOPPED, ON_SAVE_INSTANCE_STATE, ON_DESTROYED
+    }
+
+    init {
+        attachBaseContext(mBaseApplication)
+        mActivityLifecycleCallbackRecorder = ActivityLifecycleCallbackRecorder()
+        startRecordLifecycles()
+    }
+
+    fun setActivityLifecycleCallbackRecorder(activityLifecycleCallbackRecorder: ActivityLifecycleCallbackRecorder?) {
+        mActivityLifecycleCallbackRecorder = activityLifecycleCallbackRecorder
+    }
+
+    var isRecordActivityLifecycle: Boolean
+        get() = mRecord
+        set(record) {
+            if (record.also { this.mRecord = it }) {
+                startRecordLifecycles()
+            } else {
+                stopRecordLifecycles()
+            }
+        }
+
+    @SuppressLint("MissingSuperCall")
```

Comment:
Do we truly not want to be calling super here and in the methods below?

Reply:
yes it's required here because we need here our custom logic so.